### PR TITLE
doc: add updates for new noarch python syntax from CFEP-25

### PR DESCRIPTION
**Pull request by beckermr**
PR Checklist:

- [x] note any issues closed by this PR with [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] if you are adding a new page under `docs/` or `community/`, you have added it to the sidebar in the corresponding `_sidebar.json` file
- [x] put any other relevant information below

This PR adds doc updates for the new noarch python syntax.

### CODE REVIEW

**Comment by beckermr** [comment]
@conda-forge/core this one is ready to go!

File: docs/maintainer/knowledge_base.md [hunk 1]
```diff
@@ -1480,6 +1480,31 @@ In order to qualify as a noarch python package, all of the following criteria mu
   in the `build` section of `meta.yaml`
 - No activate scripts
 
+All recipes employing `noarch: python` should use the `python_min` variable per the following example:
+
+```yaml title="recipe/meta.yaml"
+name: package
```
Comment (beckermr) [inline docs/maintainer/knowledge_base.md:1486]:
```suggestion
{% set python_min = python_min|default("0.1a0") %}

name: package
```

**Comment by beckermr** [comment]
@conda-forge/core any comments here or shall I merge?

File: docs/maintainer/knowledge_base.md (hunk 1 above)
Comment (mattwthompson) [reply docs/maintainer/knowledge_base.md:1488]: Could you elaborate on why this needs to be set? The CFEP says this comes from the global pinning

https://github.com/conda-forge/cfep/blob/main/cfep-25.md#specification

Comment (beckermr) [reply docs/maintainer/knowledge_base.md:1488]: This is a default that is needed for recipe parsing, but is eventually overridden by the default from the pinnings. TBH this is a workaround for a bug in conda-build, but I have not been able to track it down.

Comment (mattwthompson) [reply docs/maintainer/knowledge_base.md:1488]: Thanks for the info

Comment (mattwthompson) [reply docs/maintainer/knowledge_base.md:1488]: No longer seems to be needed https://github.com/conda-forge/conda-forge.github.io/pull/2360/files#diff-f233d562a21155b034d2aa5f9d8a8148cc0952b24c34f8e0e6cf0a1a4e241acfL1486-L1487

**Pull request merged**